python/cuspatial/cuspatial/geometry/geocolumn.py

```diff
@@ -0,0 +1,471 @@
+                    self.input_lengths += [1]
+                    mline_size = (
+                        buffers.lines.mlines[ml_index * 2 + 1]
+                        - 1
+                        - buffers.lines.mlines[ml_index * 2]
+                    )
+                    self.input_types.extend(repeat("l", mline_size))
+                    self.input_lengths.extend(repeat(1, mline_size))
+            else:
+                self.input_types.extend(repeat("l", len(buffers.lines)))
+                self.input_lengths.extend(repeat(1, len(buffers.lines)))
+        if buffers.polygons is not None:
+            if len(buffers.polygons.mpolys) > 0:
+                self.input_types.extend(
+                    repeat("poly", buffers.polygons.mpolys[0])
+                )
+                self.input_lengths.extend(
+                    repeat(1, buffers.polygons.mpolys[0])
+                )
+                for mp_index in range(len(buffers.polygons.mpolys) // 2):
+                    mpoly_size = (
+                        buffers.polygons.mpolys[mp_index * 2 + 1]
+                        - buffers.polygons.mpolys[mp_index * 2]
+                    )
+                    self.input_types.extend(["mpoly"])
+                    self.input_lengths.extend([mpoly_size])
+                    self.input_types.extend(repeat("poly", mpoly_size))
+                    self.input_lengths.extend(repeat(1, mpoly_size))
+            else:
+                self.input_types.extend(repeat("poly", len(buffers.polygons)))
+                self.input_lengths.extend(repeat(1, len(buffers.polygons)))
+
+    def copy(self, deep=True):
+        return type(self)(
+            {
+                "input_types": self.input_types.copy(),
+                "input_lengths": self.input_lengths.copy(),
+            }
+        )
+
+
+class GeoColumn(NumericalColumn):
+    """
+    Parameters
+    ----------
+    data : A GeoArrowBuffers object
+    meta : A GeoMeta object (optional)
+
+    Notes
+    -----
+    The GeoColumn class subclasses `NumericalColumn`. Combined with
+    `_copy_type_metadata`, this assures support for existing cudf algorithms.
+    """
+
+    def __init__(
+        self,
+        data: GeoArrowBuffers,
+        meta: GeoMeta = None,
+        shuffle_order: cudf.Index = None,
+    ):
+        base = cudf.Series(cudf.RangeIndex(0, len(data)))._column.data
+        super().__init__(base, dtype="int64")
+        self._geo = data
+        if meta is not None:
+            self._meta = meta
+        else:
+            self._meta = GeoMeta(data)
+        if shuffle_order is not None:
+            self._data = shuffle_order
+
+    def __iter__(self):
+        self._iter_index = 0
+        return self
+
+    def __next__(self):
+        if self._iter_index >= len(self):
+            raise StopIteration
+        result = self[self._iter_index]
+        self._iter_index = self._iter_index + 1
+        return result
```

Comment:
If we need these functions implemented than something is off. This allows us to fall into some really slow code paths that we should rather throw an Exception for than allow.

Reply:
I was really just using it for an internal function, after `to_host`, to explode the `GeoColumn` into the necessary `gpuGeometries` (which I'm working to get rid of today) for moving back to GeoPandas. I've modified `to_geopandas` to iterate on the length instead of using a list expansion, and gotten rid of this iterator.